g4f/__init__.py

```diff
@@ -43,12 +57,20 @@ def create(
         if provider.needs_auth:
             kwargs['auth'] = auth
 
-        if not provider.supports_stream and stream:
-            raise Exception(
-                f'ValueError: {provider.__name__} does not support "stream" argument')
-
-        if logging:
-            print(f'Using {provider.__name__} provider')
-
         result = provider.create_completion(model.name, messages, stream, **kwargs)
         return result if stream else ''.join(result)
+
+    @staticmethod
+    async def create_async(
+        model    : Union[models.Model, str],
+        messages : list[dict[str, str]],
+        provider : Union[type[BaseProvider], None] = None,
+        **kwargs
+    ) -> str:
+        
+        model, provider = get_model_and_provider(model, provider, False)
+
+        if not issubclass(provider, AsyncProvider):
```

Comment:
You can create providers that inherit directly from BaseProvider too here, just need to handle the case.
Ofc you are more familiar with the codebase, so if you are planning to dismiss the use of purely synchronous providers, the change makes sense.

Reply:
When we use BaseProvider here, it will still executed synchronous like in create.
I changed already all working providers to "AsyncProvider".

Reply:
Thank you for clarifying, all good then!